worker/uniter/uniter.go

```diff
@@ -92,28 +98,36 @@ type Uniter struct {
 	updateStatusAt TimedSignal
 }
 
+// UniterParams hold all the necessary parameters for a new Uniter.
+type UniterParams struct {
+	St                 *uniter.State
+	UnitTag            names.UnitTag
+	LeadershipManager  coreleadership.LeadershipManager
+	DataDir            string
+	HookLock           *fslock.Lock
+	MetricsTimer       *timerChooser
+	UpdateStatusSignal TimedSignal
```

Comment:
Did we actually want/need this? I thought we could do it entirely within the loop func.

(To be clear, I'm quite happy to have the timing managed by some type; but I don't think it's instance-scoped, I think it's only relevant within the abide loop.)